### PR TITLE
Renaming tool name to `opcap`

### PR DESCRIPTION
**Pull request by yashoza19**
Renaming project name to `opcap` from `capabilities-tool`. Also changing subcommand to use `level-1` instead of `index capabilities`.

Fixes https://github.com/opdev/capabilities-tool/issues/11

Signed-off-by: Yash Oza [yoza@gmail.com](mailto:yoza@gmail.com)

### CODE REVIEW

File: cmd/level-1/command.go [hunk 1]
```diff
@@ -53,7 +53,7 @@ func NewCmd() *cobra.Command {
 		"filter by the Package names which are like *package-name*. Required for Operator Clean-up")
 	cmd.Flags().StringVar(&flags.BundleName, "bundle-name", "",
 		"filter by the Bundle names which are like *bundle-name*")
-	cmd.Flags().StringVar(&flags.FilterBundle, "bundle-image", "",
+	cmd.Flags().StringVar(&flags.BundleImage, "bundle-image", "",
```
Comment (acmenezes) [inline cmd/level-1/command.go:56]: This change is not related to the tool renaming. At doesn't appear to be. That's fine for now but would be nice to have it in a separate commit but possibly on the same PR.

File: openshift/capabilities-tool-job.yaml [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
+  name: opcap-job
 spec:
   template:
     spec:
       containers:
-      - name: capabilities-tool
-        image: quay.io/opdev/capabilities-tool:v0.2.7
-        args: ["index","capabilities","--container-engine","podman","--output-path","/opt/capabilities-tool","--bundle-image","$(bundle_image)","--bucket-name","$(bucket_name)","--bundle-name","$(bundle_name)"]
+      - name: opcap
+        image: quay.io/opdev/opcap:v4.9
```
Comment (acmenezes) [inline openshift/capabilities-tool-job.yaml:10]: Is 4.9 to align with OpenShift? I think we should keep versions on 0.x.y until we get this actually in OpenShift. That really suggests two things one that we are in stable stream or that we are already part of OCP release. Normally we would have an upstream project with a proper version and a downstream to be distributed with OCP if we will. The downstream one would be the tool with OCP numbers. Another thought is that we could tie the orchestrator/runner/operator to OCP versions but allow opcap to evolve independently and just set the supported versions of opcap on the orchestration tool according to what has been tested.

**Review by acmenezes — CHANGES_REQUESTED** [review]
Everything seems good. Just one suggestion on separating multiple commits in one PR when the change requires that and a note on the image version. That's open to discussion but I don't think we should follow OCP versions on an upstream project.

File: openshift/capabilities-tool-job.yaml (hunk 2 above)
Comment (yashoza19) [reply openshift/capabilities-tool-job.yaml:10]: @mrhillsman

File: cmd/level-1/command.go (hunk 1 above)
Comment (yashoza19) [reply cmd/level-1/command.go:56]: Apologies for changing flags in this PR. I will keep in mind to have a separate PR for each change moving forward

Comment (yashoza19) [reply cmd/level-1/command.go]: Done. Removed unrelated changes from this PR.

File: cmd/level-1/command.go [hunk 3]
```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package capabilities
+package level_1
```
Comment (mrhillsman) [inline cmd/level-1/command.go:15]: package one instead of level_1

File: cmd/main.go [hunk 4]
```diff
@@ -15,28 +15,21 @@
...
-			"of the report, image name and date. " +
-			"(E.g. `testdata/report/bundles_quay.io_operatorhubio_catalog_latest_2021-04-22.xlsx`)" +
-			"For further information use the --help and check : https://github.com/operator-framework/audit",
+		Use:   "opcap",
+		Short: "A tool to audit operator bundles and run checks for different operator capability levels",
+		Long:  "",
 	}
 
-	rootCmd.AddCommand(index.NewCmd())
+	rootCmd.AddCommand(level_1.NewCmd())
```
Comment (mrhillsman) [inline cmd/main.go:32]: one.NewCmd

File: openshift/capabilities-tool-job.yaml (hunk 2 above)
Comment (mrhillsman) [reply openshift/capabilities-tool-job.yaml:10]: I don't think our tag specifically speaks to being part of OpenShift but I understand the stable stream argument. I like the latter suggestion @acmenezes makes. We want to tie the overall audit to a version of OpenShift as it is what we are stating to have successfully run the tooling against and any audit of said tooling against a version of OpenShift that it does not match has not been tested. That being said I think the runner and eventually the operator is the appropriate place to put the tags.

File: openshift/capabilities-tool-job.yaml [hunk 5]
```diff
@@ -1,14 +1,14 @@
...
+        args: ["level-1","--container-engine","podman","--output-path","/opt/capabilities-tool","--bundle-image","$(bundle_image)","--bucket-name","$(bucket_name)","--bundle-name","$(bundle_name)"]
```
Comment (mrhillsman) [inline openshift/capabilities-tool-job.yaml:11]: one versus level-1

File: pkg/actions/extract_index.go [hunk 6]
```diff
@@ -39,10 +39,10 @@ func ExtractIndexDB(image string, containerEngine string) error {
 	}
 
 	// Extract
-	command = exec.Command(containerEngine, "cp", fmt.Sprintf("%s:/database/index.db", catalogIndex), "./output/")
+	command = exec.Command(containerEngine, "cp", fmt.Sprintf("%s:/database/level-1.db", catalogIndex), "./output/")
 	_, err = pkg.RunCommand(command)
 	if err != nil {
-		return fmt.Errorf("unable to extract the image for index.db %s : %s", image, err)
+		return fmt.Errorf("unable to extract the image for level-1.db %s : %s", image, err)
```
Comment (mrhillsman) [inline pkg/actions/extract_index.go:45]: one versus level-1

File: pkg/actions/extract_index.go [hunk 7]
```diff
@@ -39,10 +39,10 @@ func ExtractIndexDB(image string, containerEngine string) error {
 	}
 
 	// Extract
-	command = exec.Command(containerEngine, "cp", fmt.Sprintf("%s:/database/index.db", catalogIndex), "./output/")
+	command = exec.Command(containerEngine, "cp", fmt.Sprintf("%s:/database/level-1.db", catalogIndex), "./output/")
```
Comment (mrhillsman) [inline pkg/actions/extract_index.go:42]: one versus level-1

File: pkg/actions/extract_index.go [hunk 8]
```diff
@@ -20,10 +20,10 @@ import (
 
 	log "github.com/sirupsen/logrus"
 
-	"capabilities-tool/pkg"
+	"opcap/pkg"
 )
 
-const catalogIndex = "audit-catalog-index"
+const catalogIndex = "audit-catalog-level-1"
```
Comment (mrhillsman) [inline pkg/actions/extract_index.go:26]: one versus level-1

**Review by mrhillsman — CHANGES_REQUESTED** [review]
directory cmd > level-1 should be cmd > one per our discussion

File: openshift/capabilities-tool-job.yaml (hunk 2 above)
Comment (yashoza19) [reply openshift/capabilities-tool-job.yaml:10]: >Another thought is that we could tie the orchestrator/runner/operator to OCP versions but allow opcap to evolve independently and just set the supported versions of opcap on the orchestration tool according to what has been tested.


@acmenezes Makes sense, I'm going to change the opcap version to 0.0.1.

File: pkg/actions/extract_index.go [hunk 9]
```diff
@@ -20,10 +20,10 @@ import (
 
 	log "github.com/sirupsen/logrus"
 
-	"capabilities-tool/pkg"
+	"opcap/pkg"
 )
 
-const catalogIndex = "audit-catalog-index"
+const catalogIndex = "audit-catalog-one"
```
Comment (mrhillsman) [inline pkg/actions/extract_index.go:26]: index here should be fine

File: pkg/actions/extract_index.go [hunk 10]
```diff
@@ -39,10 +39,10 @@ func ExtractIndexDB(image string, containerEngine string) error {
 	}
 
 	// Extract
-	command = exec.Command(containerEngine, "cp", fmt.Sprintf("%s:/database/index.db", catalogIndex), "./output/")
+	command = exec.Command(containerEngine, "cp", fmt.Sprintf("%s:/database/one.db", catalogIndex), "./output/")
```
Comment (mrhillsman) [inline pkg/actions/extract_index.go:42]: index here should be fine

File: pkg/actions/extract_index.go [hunk 11]
```diff
@@ -39,10 +39,10 @@ func ExtractIndexDB(image string, containerEngine string) error {
...
 	_, err = pkg.RunCommand(command)
 	if err != nil {
-		return fmt.Errorf("unable to extract the image for index.db %s : %s", image, err)
+		return fmt.Errorf("unable to extract the image for one.db %s : %s", image, err)
```
Comment (mrhillsman) [inline pkg/actions/extract_index.go:45]: index here should be fine

File: pkg/actions/extract_index.go (hunk 9 above)
Comment (mrhillsman) [inline pkg/actions/extract_index.go:26]: index here should be fine

**Review by mrhillsman — CHANGES_REQUESTED** [review]

File: pkg/actions/extract_index.go (hunk 10 above)
Comment (yashoza19) [reply pkg/actions/extract_index.go:42]: Will be removing this action since it is not being used opcap anymore.

File: pkg/actions/extract_index.go (hunk 11 above)
Comment (yashoza19) [reply pkg/actions/extract_index.go:45]: Will be removing this action since it is not being used opcap anymore.

File: pkg/actions/extract_index.go (hunk 9 above)
Comment (yashoza19) [reply pkg/actions/extract_index.go:26]: Will be removing this action since it is not being used opcap anymore.

**Comment by madorn** [comment]
Update the [Dockerfile](https://github.com/yashoza19/capabilities-tool-1/blob/rename_tool/Dockerfile) to ensure bin and container user reflect new name

**Comment by madorn** [comment]
LGTM

**Review by mrhillsman — APPROVED** [review]

**Review by acmenezes — APPROVED** [review]
/lgtm
